app/src/androidTest/java/com/github/sdpcoachme/database/MockDatabase.kt

```diff
@@ -209,41 +212,55 @@ open class MockDatabase: Database {
             val msg = Message(sender = chat.participants[0], content = "test onChange method", timestamp = LocalDateTime.now().toString())
             chat = chat.copy(id = this.chatId , messages = chat.messages + msg)
         } else {
-            this.chatId = chatId
-            chat = chat.copy(id = chatId)
-            this.onChange = onChange
-            numberOfAddChatListenerCalls++
+            if (chatId.startsWith("@@event")) {
+                groupChat = groupChat.copy(id = chatId)
+                this.onChange(groupChat)
+                return
+            } else {
+                this.chatId = chatId
+                chat = chat.copy(id = chatId)
+                this.onChange = onChange
+            }
         }
         this.onChange(chat)
     }
 
-    override fun getChatContacts(email: String): CompletableFuture<List<UserInfo>> {
-        return CompletableFuture.completedFuture(listOf(toUser))
+    override fun getContactRowInfo(email: String): CompletableFuture<List<ContactRowInfo>> {
+        val id = if (email < toUser.email) email+toUser.email else toUser.email+email
+        return CompletableFuture.completedFuture(listOf(
+            ContactRowInfo(id, toUser.firstName + " " + toUser.lastName, if (chat.messages.isEmpty()) Message() else chat.messages.last()),
+            ContactRowInfo(groupChat.id, "Group Chat", if (groupChat.messages.isEmpty()) Message() else groupChat.messages.last(), true)
+        ))
     }
 
+    override fun updateChatParticipants(
+        chatId: String,
+        participants: List<String>
+    ): CompletableFuture<Void> {
+        if (chatId.startsWith("@@event")) { // only group chats can be updated
+            groupChat = groupChat.copy(id = chatId, participants = participants)
+        }
+        return CompletableFuture.completedFuture(null)
+    }
 
     override fun markMessagesAsRead(chatId: String, email: String): CompletableFuture<Void> {
-        chat = Chat.markOtherUsersMessagesAsRead(chat, email)
+        if (chatId.startsWith("@@event")) {
+            groupChat = Chat.markOtherUsersMessagesAsRead(groupChat, email)
+        } else {
+            chat = Chat.markOtherUsersMessagesAsRead(chat, email)
+        }
 
         return CompletableFuture.completedFuture(null)
     }
 
     override fun removeChatListener(chatId: String) {
-        numberOfRemovedChatListenerCalls++
+        // Not necessary
```

Comment:
Should we keep this function then ?

Reply:
As MockDatabase implements Database, we need to also have an implementation for the methods so we cannot remove them